api/terraform/data_api/terraform.tf

```diff
@@ -2,10 +2,10 @@ terraform {
   required_version = ">= 0.9"
 
   backend "s3" {
-    role_arn = "arn:aws:iam::760097843905:role/platform-developer"
+    role_arn = "arn:aws:iam::756629837203:role/catalogue-developer"
 
-    bucket         = "wellcomecollection-platform-infra"
-    key            = "terraform/data_api.tfstate"
+    bucket         = "wellcomecollection-catalogue-infra-delta"
+    key            = "terraform/catalogue/api/data_api.tfstate"
```

Comment:
Just something I am picking up, we are categorising via folder structure, and I am unsure how, as we do it differently elsewhere.

is it `terraform/{repo}/{project}`? Is so should it not just be `terraform/catalogue/data_api`?

Reply:
`terraform/{repo}/{project}` is what I’m aiming for, yeah

It’s not super-duper consistent; the aim is that somebody could look at a terraform remote state key and work out where it’s defined, rather than being completely machine-readable. (In the past it’s become tricky to know what states are actively maintained/in a renamed folder/stale.)

Reply:
I'm thinking for humans when we create a new one and look how it's been done before and there's three ways, then I need to work out which one to follow.

How do you feel about keeping it we've specified above?

Reply:
Ah, sorry, I see the distinction now.

`terraform/{repo}/{project}` is *not* the idea in my head, it’s `terraform/{repo}/{path_within_repo}`.

I don’t feel strongly about it either way, but will note that we already have `terraform/catalogue/api/update_api_docs.tfstate` in the stack from the previous PR.

Reply:
I like the hierarchy @alexwlchan has introduced - it namespaces stack to project which is useful and necessary as we split up stacks.